internal/resources/app_endpoint_schema.go

```diff
@@ -0,0 +1,118 @@
+package resources
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// AppEndpointSchema defines the schema for the AppEndpoint resource.
+func AppEndpointSchema() schema.Schema {
+	return schema.Schema{
+		MarkdownDescription: "This resource allows you to manage an App Endpoint configuration for a Couchbase Capella App Service.",
+		Attributes: map[string]schema.Attribute{
+			"organization_id":  WithDescription(stringAttribute([]string{required, requiresReplace}), "The GUID4 ID of the Couchbase Capellaorganization."),
+			"project_id":       WithDescription(stringAttribute([]string{required, requiresReplace}), "The GUID4 ID of the Couchbase Capella project."),
+			"cluster_id":       WithDescription(stringAttribute([]string{required, requiresReplace}), "The GUID4 ID of the Couchbase Capella cluster."),
+			"bucket":           WithDescription(stringAttribute([]string{required, requiresReplace}), "The name of the bucket associated with this App Endpoint."),
+			"name":             WithDescription(stringAttribute([]string{required, requiresReplace}), "The name of the App Endpoint."),
+			"userXattrKey":     WithDescription(stringAttribute([]string{optional}), "The user extended attribute key for the App Endpoint."),
+			"deltaSyncEnabled": WithDescription(boolAttribute(optional), "Whether delta sync is enabled for this App Endpoint."),
```

Comment:
if user does not provide `userXattrKey` or `deltaSyncEnabled`, does the backend return some value ? if yes then it should also be `computed`